didcomm_messaging/v1/messaging.py

```diff
@@ -84,7 +84,11 @@ async def did_to_target(
             self.vm_to_v1_kid(crypto, doc, recip) for recip in target.recipient_keys
         ]
         routing_keys = [
-            self.vm_to_v1_kid(crypto, doc, routing_key)
+            crypto.public_key_to_v1_kid(
+                crypto.verification_method_to_public_key(
+                    VerificationMethod(id=routing_key, type="Ed25519VerificationKey2020", controller=routing_key.split("#")[0], public_key_multibase=routing_key.split("#")[1])
+                )
+            )
```

Comment:
This change seems like it might be brittle; can you give some examples of the values you were seeing and how this solution fixed it?

Reply:
Certainly! An example of a routing_key that I was seeing was: 
```
did:key:z6Mkkb3BqC1jqe91dmK4PzTQpgosXRZCT4GDWq933DeeuNmm#z6Mkkb3BqC1jqe91dmK4PzTQpgosXRZCT4GDWq933DeeuNmm
```

Reply:
That particular key happened to be what the Paradym app was sending me, but I was seeing similar things with Holdr+ in previous tests.
Paradym's DIDDoc:
```json
{
                "@context": [
                    "https://w3id.org/did/v1"
                ],
                "id": "did:peer:1zQmQMe5kPoAWgCN6iSeeABA5HfqZq62zB7e2Z8RspPHHtQK",
                "service": [
                    {
                        "id": "#inline-0",
                        "serviceEndpoint": "https://mediator.paradym.id/didcomm",
                        "type": "did-communication",
                        "priority": 0,
                        "recipientKeys": [
                            "#key-1"
                        ],
                        "routingKeys": [
                            "did:key:z6Mkkb3BqC1jqe91dmK4PzTQpgosXRZCT4GDWq933DeeuNmm#z6Mkkb3BqC1jqe91dmK4PzTQpgosXRZCT4GDWq933DeeuNmm"
                        ]
                    }
                ],
                "authentication": [
                    {
                        "id": "#key-1",
                        "type": "Ed25519VerificationKey2018",
                        "controller": "#id",
                        "publicKeyBase58": "FrZz8ikxtWdTjWxeGQ8q9NgNAi6MqF2Xap7LtvHnmDW"
                    }
                ],
                "keyAgreement": [
                    {
                        "id": "#key-2",
                        "type": "X25519KeyAgreementKey2019",
                        "controller": "#id",
                        "publicKeyBase58": "taZeuFnPgzcGK35rFwh5uFZFd68smurVBqGKycNwaYm"
                    }
                ]
            }
```

Reply:
The exception I was getting before this fix was as follows:
```
Traceback (most recent call last):
  File "/srv/http/login-service/auth/env/lib/python3.11/site-packages/aiohttp/web_protocol.py", line 433, in _handle_request
    resp = await request_handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/srv/http/login-service/auth/env/lib/python3.11/site-packages/aiohttp/web_app.py", line 504, in _handle
    resp = await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/srv/http/login-service-wolkins/auth/didcomm.py", line 296, in dump_body_handler
    packy = await dmp.pack(
            ^^^^^^^^^^^^^^^
  File "/srv/http/login-service/auth/env/lib/python3.11/site-packages/didcomm_messaging/v1/messaging.py", line 252, in pack
    return await self.dmp.pack(
           ^^^^^^^^^^^^^^^^^^^^
  File "/srv/http/login-service/auth/env/lib/python3.11/site-packages/didcomm_messaging/v1/messaging.py", line 168, in pack
    target = await self.did_to_target(crypto, resolver, to)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/srv/http/login-service/auth/env/lib/python3.11/site-packages/didcomm_messaging/v1/messaging.py", line 86, in did_to_target
    routing_keys = [
                   ^
  File "/srv/http/login-service/auth/env/lib/python3.11/site-packages/didcomm_messaging/v1/messaging.py", line 87, in <listcomp>
    self.vm_to_v1_kid(crypto, doc, routing_key)
  File "/srv/http/login-service/auth/env/lib/python3.11/site-packages/didcomm_messaging/v1/messaging.py", line 65, in vm_to_v1_kid
    doc.dereference_as(VerificationMethod, ref)
  File "/srv/http/login-service/auth/env/lib/python3.11/site-packages/pydid/resource.py", line 128, in dereference_as
    resource = self.dereference(reference)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/srv/http/login-service/auth/env/lib/python3.11/site-packages/pydid/doc/doc.py", line 132, in dereference
    raise IDNotFoundError("ID {} not found in document".format(reference))
pydid.doc.doc.IDNotFoundError: ID did:key:z6MkvLvjZVc9jWJVs8M3qNZeRsnXXZGFjzcRJDH9vvdJMgHB#z6MkvLvjZVc9jWJVs8M3qNZeRsnXXZGFjzcRJDH9vvdJMgHB not found in document
```

Reply:
The problem with the old code was that it was trying to dereference the URL in the context of the DID document containing the DIDComm service endpoint (the `doc` parameter). Rather than dereferencing in the current doc, we need to resolve a new doc and dereference against that.